### DISCUSSION TITLE
pm2 does not work on cluster mode, any workaround?

### DISCUSSION THREAD

**Discussion by ironydips** [oven-sh/bun#10614] [Q&A]
When running `pm2 start index.js -i 3 --interpreter=bun ` instead of running in cluster mode, it works in fork mode. 

Any workaround to use all the cores of cpus? any alternatives or solution to this?

bun -v 1.1.6

and pm2 has been installed using `bun i -g pm2`

<img width="951" alt="Screenshot 2024-04-28 at 9 47 20 PM" src="https://github.com/oven-sh/bun/assets/2323273/b9b7deac-fb3a-4522-9917-3db87c73a806">

**Comment by ironydips** [comment]
@liudonghua123 can you help here pls.

**Reply by liudonghua123** [reply]
Hi, could you share more details about this problem. What's the content of index.js? What's your expected behavior?

**Reply by ironydips** [reply]
Sure. I have to run the  main file of our app (index.js) in cluster mode, like it happens when we run on node environment.

We have to run multiple instances of our backend application through pm2 so that we can utilise all cores of Cpus, which we have been already doing on our node env, however when switching to Bun environment, pm2 command (pm2 start index.js -i 3) does not run application in cluster mode, rather as you can see in above screenshot it runs on FORK mode, means as different application. 

index.js file creates and listens server on port:8001, now if it runs on fork mode, it will try to open port 3 times, which will get rejected after 1st attempt.

**Reply by Igloczek** [reply]
That's a bummer, also looking for a solution, not even necessarily with PM2, anything that can make the app work on multiple threads, handle the auto-balancing and will keep the app alive, even when some process fails.

**Reply by liudonghua123** [reply]
I tried to run the example code of https://nodejs.org/api/cluster.html#cluster with the latest bun 1.1.8. But it seems doesn't support `node:cluster` yet.

And there is already issue tracking this feature. Maybe need to wait https://github.com/oven-sh/bun/issues/2428 completed to make pm2 cluster mode working.

```
> bun index.mjs
Primary 2760 is running
14 |   warned.add(feature), console.warn(new NotImplementedError(feature, issue));
15 | }, $;
16 |
17 | class NotImplementedError extends Error {
18 |   code;
19 |   constructor(feature, issue) {
                               ^
NotImplementedError: node:cluster is not yet implemented in Bun. Track the status & thumbs up the issue: https://github.com/oven-sh/bun/issues/2428
 code: "ERR_NOT_IMPLEMENTED"

      at new NotImplementedError (internal:shared:19:27)
      at internal:shared:2:69
      at fork (node:cluster:25:13)
      at D:\code\node\playground\cluster\index.mjs:13:5

Bun v1.1.8 (Windows x64)
```

**Comment by re-thc** [comment]
Does setting `reusePort: true` in the HTTP server config fix it? There is a workaround for cluster if it's on Linux.
